lib/api/CFieldDataTyper.cc

```diff
@@ -422,6 +423,19 @@ bool CFieldDataTyper::periodicPersistState(CBackgroundPersister& persister) {
     return true;
 }
 
+bool CFieldDataTyper::periodicPersistStateInForeground(CPersistenceManager& persistenceManager) {
+    LOG_DEBUG(<< "Periodic persist typer state");
```

Comment:
This doesn't have any chaining logic as in
```
if (m_OutputHandler.periodicPersistStateInForeground(persister) == false) {
    return false;
}
```
Is this intentional?

Reply:
Oh I see this is in `this->persistState`. I think this warrants a comment.

Reply:
Also, I don't totally understand the logic here: this object has a pointer to a persist manager, but it is passed in here and for example in `this->persistState` it checks to see if the member variable persist is busy rather than the supplied `persistenceManager` object.